### ISSUE TITLE
feature request: show random orphan note

### ISSUE THREAD

**Issue by hauselin** [erichalldev/obsidian-smart-random-note#5]
Great plugin! I think it'd be cool to be able to show a randomly chosen orphan note. Not sure how difficult/memory-intensive it would be, but maybe check out [this plugin](https://github.com/Vinzent03/find-unlinked-files).

**Comment by erichalldev** [comment]
Thanks! This is similar to #4, I think. Do you consider an orphaned note to be unlinked AND untagged? Or just unlinked? Either way, I don't think it'd be memory-intensive since the heavy lifting is already done by Obsidian with the `metadataCache`. Thanks for referencing that plugin, it looks like how I'd approach getting unlinked files. I'll look into what it would take to make this happen.

**Comment by hauselin** [comment]
Good to know it won't be memory intensive. I'd prefer orphan notes to be simply unlinked notes—even if a note has tags, but as long as it isn't linked to another note, it's considered an orphan. I probably have many orphan notes and I'm hoping to use your plugin to link these orphan notes, so ultimately I won't have any orphan notes. So maybe the plugin should also catch situations where there aren't any orphan notes—maybe congratulate the user ;)

Thanks! I'd love to contribute but I'm not that good with TypeScript yet, so perhaps in the future I'll work on this plugin too.

**Comment by rklasen** [comment]
For me notes can be either unlinked, untagged or both. Ideally I'd be able to include or exclude them individually from the pool of random notes.

**Comment by erichalldev** [comment]
Thanks for the input! I think it's valuable to allow for finding unlinked and untagged notes separately or together.

I'm trying to think of a way to combine the "open random note from search" functionality with opening untagged and/or unlinked notes functionality. That way you could choose to run a search then open an untagged or unlinked note from the results. I'm trying to avoid creating a GUI for as long as possible since they add so much more complexity but this might require one. 

One more question, would you consider an unlinked not to have no links and no backlinks, or just no backlinks? The way I define orphaned note is one that would appear as a node with no connections on the graph, so no links and no backlinks.

**Comment by hauselin** [comment]
For me, unlinked notes are truly unlinked. If there are backlinks, the notes are linked.

**Comment by rklasen** [comment]
I agree with @hauselin , if a note is linked to, I would not consider it unlinked. Especially since, as you said @erichalldev , I shows up in the graph view as linked.

**Comment by erichalldev** [comment]
Thanks for the input @hauselin and @rklasen. This will be the next feature I implement. In my last release I laid the plumbing to make GUIs easier to implement so that won't be as big of a problem as I originally thought.

**Comment by liutiming** [comment]
this would be a great feature :)

**Comment by erichalldev** [comment]
I haven't forgotten about this! I've spent a quite bit of time away from developing this plugin but I'm getting back to it.
